### PR TITLE
Fix for Scispark 177: Write intermediate data to HDFS while running GTG

### PR DESCRIPTION
**Pull request by sujen1412**

### CODE REVIEW

File: src/test/python/MCSVerification.py [hunk 1]
```diff
@@ -79,9 +79,12 @@ def _run_scispark_implementation():
...
         sparkSubmitStr = 'spark-submit target/scala-2.11/SciSpark.jar'
         subprocess.call(sparkSubmitStr, shell=True)
-        cpTextFilesStr = 'cp MCCEdges.txt ' + workingDir + '/scisparkGTG/textFiles'
+        outputDirectory = "output"
+        resultDir = max([os.path.join(outputDirectory,d) for d in os.listdir(outputDirectory)], key=os.path.getmtime)
+
+        cpTextFilesStr = 'cp {}/MCCEdges.txt ' + workingDir + '/scisparkGTG/textFiles'.format(resultDir)
         subprocess.call(cpTextFilesStr, shell=True)
-        cpTextFilesStr = 'cp MCCNodesLines.json ' + workingDir + '/scisparkGTG/textFiles'
+        cpTextFilesStr = 'cp {}/MCCNodesLines.json ' + workingDir + '/scisparkGTG/textFiles'.format(resultDir)
```
Comment (kwhitehall) [inline src/test/python/MCSVerification.py]: - cp line is breaking. As you are going the format option, consider something like 
  cpTextFilesStr = 'cp {}/MCCEdges.txt {}/scisparkGTG/textFiles'.format(resultDir, workingDir)
- the file name MCCNodesLines.json has been changed to MCCNodes.json. Please update through out the code.

File: src/main/scala/org/dia/algorithms/mcc/GTGRunner.scala [hunk 2]
```diff
@@ -245,32 +246,10 @@ class GTGRunner(
...
-
-    val fw = new PrintWriter("MCCEdges.txt")
-    fw.write(MCCEdgeList.toString())
-    fw.close()
-  }
-
   def run(): Unit = {
 
     logger.info("Starting MCC")
+    val outputDir = MCSUtils.testHDFSWrite(this.outputDir)
```
Comment (kwhitehall) [inline src/main/scala/org/dia/algorithms/mcc/GTGRunner.scala]: In terms of naming convention, please rename this function. In the codebase functions beginning with 'test' are found in the test dir.

File: src/main/scala/org/dia/algorithms/mcc/MCCOps.scala [hunk 3]
```diff
@@ -93,7 +96,8 @@ object MCCOps {
       partition: (Int, Iterable[MCCEdge]),
       currentIteration: Int,
       minGraphLength: Int,
-      bucketSize: Int): (Int, (Iterable[MCCEdge], mutable.MutableList[mutable.HashSet[String]])) = {
```
Comment (kwhitehall) [inline src/main/scala/org/dia/algorithms/mcc/MCCOps.scala:96]: update the docString please

File: src/main/scala/org/dia/algorithms/mcc/MCCOps.scala [hunk 4]
```diff
@@ -241,6 +247,23 @@ object MCCOps {
     return (edges, visitedNodes)
   }
```
Comment (kwhitehall) [inline src/main/scala/org/dia/algorithms/mcc/MCCOps.scala:258]: Add doc string.

**Review by kwhitehall — CHANGES_REQUESTED** [review]

File: src/test/python/MCSVerification.py (hunk 1 above)
Comment (sujen1412) [reply src/test/python/MCSVerification.py]: Yes I have corrected that through out.

File: src/main/scala/org/dia/algorithms/mcc/GTGRunner.scala (hunk 2 above)
Comment (sujen1412) [reply src/main/scala/org/dia/algorithms/mcc/GTGRunner.scala]: Ack'ed

**Review by kwhitehall — APPROVED** [review]

**Comment by kwhitehall** [comment]
LGTM. Thanks for addressing the issues @sujen1412!

**Comment by kwhitehall** [comment]
@sujen1412 the Python verification part of this PR is broken. As the PR #179 will have merge conflicts here, these will be resolved during that merge. In the future, please be certain to ensure that changes to variable names, files, e.g. propagate through all of the code.

**Pull request merged**